pages/my-library/index.js

```diff
@@ -2,28 +2,97 @@ import Header from "@/components/Header/Header";
 import BookList from "@/components/BookList/BookList";
 import Navigation from "@/components/Navigation/Navigation";
 import styled from "styled-components";
+import { useState } from "react";
 
 const StyledHeadline = styled.h2`
   text-align: center;
 `;
 
-export default function MyLibraryPage({ books, booksInfo }) {
-  const myLibraryBooks = books.filter((book) =>
-    booksInfo.find(
-      (bookInfo) => bookInfo.id === book.id && bookInfo.isBookmarked
-    )
-  );
+const StyledLibraryPage = styled.div`
+  width: 90%;
+  max-width: 600px;
+  margin: auto;
+`;
+
+const StyledParagraph = styled.p`
+  text-align: center;
+`;
+
+const StyledFilterButton = styled.button`
+  background-color: ${({ $isActive }) =>
+    $isActive ? "darkseagreen" : "seashell"};
+  border-radius: 4px;
+`;
+
+const StyledButtonWrapper = styled.section`
+  display: flex;
+  gap: 5px;
+`;
+
+export default function MyLibraryPage({
+  books,
+  booksInfo,
+  handleToggleBookmark,
+  handleToggleAlreadyRead,
+}) {
+  const [filter, setFilter] = useState("saved-books");
+  console.log("filter:", filter);
+  function handleFilter(newFilter) {
+    setFilter(newFilter);
+  }
+
+  const filteredBooks = books.filter((book) => {
+    if (filter === "saved-books") {
+      return booksInfo.find(
+        (bookInfo) => bookInfo.id === book.id && bookInfo.isBookmarked
+      );
+    } else if (filter === "already-read-books") {
+      return booksInfo.find(
+        (bookInfo) => bookInfo.id === book.id && bookInfo.isAlreadyRead
+      );
+    }
+  });
```

Comment:
This is more of a small untested and dryly written refactor suggestion for later - or never 😁_. 
 
 ```js
const filteredBooks = books.filter(book => {
  const bookInfo = booksInfo.find(bookInfo => bookInfo.id === book.id);
  return filter === "saved-books" ? bookInfo?.isBookmarked : bookInfo?.isAlreadyRead;
});
```



Please remove the `console.log()` before merging this branch 🧽

```suggestion
  const [filter, setFilter] = useState("saved-books");
  
  function handleFilter(newFilter) {
    setFilter(newFilter);
  }

  const filteredBooks = books.filter((book) => {
    if (filter === "saved-books") {
      return booksInfo.find(
        (bookInfo) => bookInfo.id === book.id && bookInfo.isBookmarked
      );
    } else if (filter === "already-read-books") {
      return booksInfo.find(
        (bookInfo) => bookInfo.id === book.id && bookInfo.isAlreadyRead
      );
    }
  });
```